docs/xks/developer-guide/scheduling-scaling.md

```diff
@@ -0,0 +1,254 @@
+resource requests of all Pods currently scheduled to the Node. This makes it possible to over provision as the resource request can be much larger than the actual consumption of the Pod. Efficient resource allocation is a constant battle between requesting enough resources to avoid under allocation while not
+requesting too much which would result in over allocation. A Pod may at times request more resource than any Node has capacity for, there are two possible outcomes for this situation. If the Pods resource request is less than a Nodes total available resources, a new Node will be added to the cluster. The Pod will
+however be considered unschedulable if the resource request exceeds the total resources available on a single node. In these cases either the resource request has to change or a new Node type has to be added to the cluster to cater to these needs.
+
+<img alt="Pod Scheduling" src={useBaseUrl("img/assets/xks/developer-guide/pod-scheduling.jpg")} />
+
+The resource limit defined for a Pod has no affect on the scheduling of a Pod. Limits instead comes into play for a Pod during runtime. Exceeding the resource limit for CPU and memory will have
+different affects. A Pod which exceeds the memory limit will be terminated with an out of memory error (OOM Error).  The Pod will after termination be started again, it may start to exceed the limit again which will result in another OOM error. These types of errors can either be resolved by having the application
+consume less memory alternatively increasing the memory limit. Without a memory limit a Pod would be able to continue consuming memory until the Node runs out. This would not only affect critical
+system processes that runs in the node but other Pods which may not even be able to consume the resources it requested.
+
+CPU limits should be treated slightly differently from memory limits. When memory is over consumed applications will crash when there is no place to store new data. However when CPU is over consumed throttling will occur. Applications will not immediately crash even if performance may be severely deprecated. CPU differs from memory as a resource in the sense that CPU time will be wasted if not consumed and applications required CPU time can vary a lot. Reserving CPU time that is not used is a waste if another application would benefit from using it. Setting a CPU limit for a Pod will result in artificial CPU throttling even if it would not necessarily be required. It is for this reason why a CPU limit is not enforced for Pods by default, instead it is something that should be opted into when the effects of CPU throttling is understood. It is still important to set a reasonable CPU request for Kubernetes to determine the minimal resource requirements, but CPU limits should be avoided in most cases where it is not fully understood.

```

Comment:
Do we really have to repeat "When memory is over consumed applications will crash when there is no place to store new data." in this para?

Reply:
"It is for this reason that a CPU limit..."

Reply:
It might be pertinent to mention that getting your pod throttled risks provoking internal timing bugs that are not normally visible? Can we say anything about CPU scheduling bucket size? That may help developers better understand the consequence of throttling; what scale of timeouts may occur, et c?